Sources/Perception/PerceptionRegistrar.swift

```diff
@@ -150,3 +149,16 @@ private func perceptionCheck() {
     }
   #endif
 }
+
+var isInSwiftUIStack: Bool {
+  Thread.callStackSymbols
+    .contains { $0.contains(" AttributeGraph ") }
+}
+
+var isInSwiftUIBindingGetter: Bool {
+  zip(Thread.callStackSymbols, Thread.callStackSymbols.dropFirst())
+    .contains { previousSymbol, symbol in
+      symbol.contains("SwiftUI7Binding")
+        && previousSymbol.contains("swift_readAtKeyPath")
+    }
```

Comment:
Does this work well for other checks? For some reason I don't remember seeing `Binding` in other traces where we try to detect this.

Reply:
We just have to use the task local still in TCA, like the `$store.scope` bindings, but I think it's necessary.

Reply:
Well if we backport `@Bindable` it would work for that...though for `@State` sounds like we can't detect?

Reply:
Yeah, I don't think it's possible. We could just disable runtime checks entirely if we see `swift_getAtKeyPath` in the call stack because we just can't know if it's a binding being called within `WithPerceptionTracking` or not.

Reply:
Won't that be in the trace for a normal access in a view? Or is that OK?

Reply:
Doesn't seem to. Just pushed a commit that at least has the test suite passing with one "NB" of something that we can hopefully make emit a runtime warning someday.

With these changes it is possible that you could have a view with _only_ bindings, and you would not get a warning if you forget `WithPerceptTracking`. But hopefully that is relatively rare.